src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -684,6 +699,10 @@ protected override bool UnbindOnTransactionCompletion
             }
         }
 
+        /// <summary>
+        /// Validates if federated authentication is used, Access Token used by this connection is active for the value of 'accessTokenExpirationBufferTime'.
+        /// </summary>
+        internal override bool IsAccessTokenExpired => _federatedAuthenticationInfoRequested && DateTime.FromFileTimeUtc(_fedAuthToken.expirationFileTime) < DateTime.UtcNow.AddSeconds(accessTokenExpirationBufferTime);
```

Comment:
So, here if the connection time out is greater than 600 or infinite time out is requested, it will expire the access token 10 minutes before its actual time. otherwise, it will expire for the connection time out. Am I right? If yes, there is a big gap between like 30 or 60 seconds and 10 minutes. Isn't  it going to make a problem?

Reply:
besides this question the PR looks good to me.

Reply:
No the intent is to expire access tokens only when they should expire.